libfreerdp/core/peer.c

```diff
@@ -567,16 +567,34 @@ static int peer_recv_callback(rdpTransport* transport, wStream* s, void* extra)
 			break;
 
 		case CONNECTION_STATE_LICENSING:
-			if (!license_send_valid_client_error_packet(rdp->license))
+		{
+			LicenseCallbackResult res;
+			if (!client->LicenseCallback)
 			{
-				WLog_ERR(TAG,
-				         "peer_recv_callback: CONNECTION_STATE_LICENSING - license_send_valid_client_error_packet() fail");
+				WLog_ERR(TAG, "peer_recv_callback: LicenseCallback has been removed, assuming licensing is ok (please fix your app)");
+				res = LICENSE_CB_COMPLETED;
```

Comment:
Shouldn't this be an error condition?

Reply:
Well, it could but in practice we don't really care. To be in that case, we must have an app that would have replaced, on purpose, the callback by `NULL`. Looks very unlikely.

Reply:
or like a beginners mistake ;)
anyway, not really that important.